Dockerfile

```diff
@@ -2,10 +2,13 @@
 FROM erlang:21-alpine AS scratch
 MAINTAINER Petr Gotthard <petr.gotthard@centrum.cz>
 
-RUN apk add --no-cache --virtual build-deps git make wget nodejs-npm && \
-    git clone https://github.com/gotthardp/lorawan-server.git && \
-    cd lorawan-server && \
-    make release
+RUN apk add --no-cache --virtual build-deps git make wget nodejs-npm
+
+WORKDIR /lorawan-server
```

Comment:
@citruspress This should be mkdir

Reply:
WORKDIR creates the directory if it doesn't exist and changes the working directory to that folder.

using mkdir requires repeating the "lorawan-server" directory in 3 places instead of a single place (here)...

Reply:
Yes, I am aware of this of course, but this command is not linear. If you, for some reason, insist on using WORKDIR directive, then make it next after FROM. As the execution goes it does not really matter, but for the intermediate and overall container optimization it makes a difference. This way you can return the 'make release' call back into the first RUN statement, then using WORKDIR will be of benefit.

Reply:
For container optimization when using a multistage build, the main thing you should care about is build performance. Since the runtime is in a second stage anyway. Moving 'make release' to the same layer as apk add ruins the build cache and therefor the build performance.